Build failure: policycoreutils

### Steps To Reproduce

Steps to reproduce the behavior:
1. Assume that the user runs NixOS 24.05 (Uakari)
2. add `policycoreutils` to `environment.systemPackages`
3. run `nixos-rebuild switch`

### Build log

```
@nix { "action": "setPhase", "phase": "unpackPhase" }
Running phase: unpackPhase
unpacking source archive /nix/store/plh37z34d4mh7rdw3dympspmmb39344r-policycoreutils-3.3.tar.gz
source root is policycoreutils-3.3
setting SOURCE_DATE_EPOCH to timestamp 1634826683 of file policycoreutils-3.3/setsebool/setsebool.c
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: patchPhase
substituteStream(): WARNING: '--replace' is deprecated, use --replace-{fail,warn,quiet}. (file 'po/Makefile')
substituteStream(): WARNING: pattern /usr/share doesn't match anything in file 'po/Makefile'
substituteStream(): WARNING: pattern /usr/share doesn't match anything in file 'newrole/Makefile'
@nix { "action": "setPhase", "phase": "unpackPhase" }
Running phase: unpackPhase
unpacking source archive /nix/store/plh37z34d4mh7rdw3dympspmmb39344r-policycoreutils-3.3.tar.gz
source root is policycoreutils-3.3
setting SOURCE_DATE_EPOCH to timestamp 1634826683 of file policycoreutils-3.3/setsebool/setsebool.c
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: patchPhase
substituteStream(): WARNING: '--replace' is deprecated, use --replace-{fail,warn,quiet}. (file 'po/Makefile')
substituteStream(): WARNING: pattern /usr/share doesn't match anything in file 'po/Makefile'
substituteStream(): WARNING: pattern /usr/share doesn't match anything in file 'newrole/Makefile'
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "configurePhase" }
Running phase: configurePhase
no configure script, doing nothing
@nix { "action": "setPhase", "phase": "buildPhase" }
Running phase: buildPhase
build flags: SHELL=/nix/store/087167dfxal194pm54cmcbbxsfy3cjgn-bash-5.2p26/bin/bash PREFIX=\$\(out\) SBINDIR=\$\(out\)/>
make[1]: Entering directory '/build/policycoreutils-3.3/setfiles'
gcc -g -Werror -Wall -W   -c -o setfiles.o setfiles.c
gcc -g -Werror -Wall -W   -c -o restore.o restore.c
gcc   setfiles.o restore.o  -lselinux -lsepol -o setfiles
ln -sf setfiles restorecon
gcc -g -Werror -Wall -W   -c -o restorecon_xattr.o restorecon_xattr.c
gcc   restorecon_xattr.o restore.o  -lselinux -lsepol -o restorecon_xattr
make[1]: Leaving directory '/build/policycoreutils-3.3/setfiles'
make[1]: Entering directory '/build/policycoreutils-3.3/load_policy'
gcc -Werror -Wall -W  -DUSE_NLS -DLOCALEDIR="\"/nix/store/7lx81bvcvhwsp87fw53pdxpm3gawgmz9-policycoreutils-3.3/share/lo>
make[1]: Leaving directory '/build/policycoreutils-3.3/load_policy'
make[1]: Entering directory '/build/policycoreutils-3.3/newrole'
gcc -Werror -Wall -W -DVERSION=\"3.3\" -DUSE_NLS -DLOCALEDIR="\"/nix/store/7lx81bvcvhwsp87fw53pdxpm3gawgmz9-policycoreu>
newrole.c: In function 'parse_command_line_arguments':
newrole.c:962:23: error: assignment discards 'const' qualifier from pointer target type [-Werror=discarded-qualifiers]
  962 |         if (!(new_con = context_str(context))) {
      |                       ^
cc1: all warnings being treated as errors
make[1]: *** [<builtin>: newrole.o] Error 1
make[1]: Leaving directory '/build/policycoreutils-3.3/newrole'
make: *** [Makefile:4: all] Error 1

```

### Additional context

Add any other context about the problem here.

### Notify maintainers

@alyssais @fpletz @Ma27 

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[root@nixos:/etc/nixos]# nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.69, NixOS, 24.05 (Uakari), 24.05.19700101.dirty`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.1`
 - channels(root): `"nixos-23.11"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/os-specific/linux/policycoreutils/default.nix
+++ b/pkgs/os-specific/linux/policycoreutils/default.nix
@@ -2,12 +2,12 @@
 
 stdenv.mkDerivation rec {
   pname = "policycoreutils";
-  version = "3.3";
+  version = "3.6";
   inherit (libsepol) se_url;
 
   src = fetchurl {
     url = "${se_url}/${version}/policycoreutils-${version}.tar.gz";
-    sha256 = "0y0hl32b2ks7r0fhbx3k2j1gqqms5aplyasjs3fz50caxl6096a1";
+    hash = "sha256-p2rEMepAo1qDFkzpAHkJwcbBL9EFZif2IhROSnBcCiw=";
   };
 
   postPatch = ''
@@ -15,8 +15,6 @@ stdenv.mkDerivation rec {
     substituteInPlace po/Makefile \
        --replace /usr/bin/install install --replace /usr/share /share
     substituteInPlace newrole/Makefile --replace /usr/share /share
-
-    sed -i -e '39i#include <crypt.h>' run_init/run_init.c
   '';
 
   nativeBuildInputs = [ gettext ];

